lib/src/boilerplate_suggestors/advanced_props_and_state_class_migrator.dart

```diff
@@ -28,11 +29,51 @@ import 'boilerplate_utilities.dart';
 class AdvancedPropsAndStateClassMigrator extends GeneralizingAstVisitor
     with AstVisitingSuggestorMixin
     implements Suggestor {
+  final bool shouldMigrateCustomClassAndMixins;
+
+  AdvancedPropsAndStateClassMigrator(
+      {this.shouldMigrateCustomClassAndMixins = false});
+
   @override
-  visitClassDeclaration(ClassDeclaration node) {
+  void visitClassDeclaration(ClassDeclaration node) {
     super.visitClassDeclaration(node);
 
     if (!shouldMigrateAdvancedPropsAndStateClass(node)) return;
+
+    final extendsFromCustomClass = !extendsFromUiPropsOrUiState(node);
+    final hasMixins = node.withClause != null;
+
+    // Don't operate if the props class uses mixins and extends a custom class,
+    // unless the flag has been set.
+    if (hasMixins &&
+        extendsFromCustomClass &&
+        !shouldMigrateCustomClassAndMixins) return;
+
+    final className = stripPrivateGeneratedPrefix(node.name.toSource());
```

Comment:
Avoid using `toSource()` when there's an alternative available that's more precise/predictable
```suggestion
    final className = stripPrivateGeneratedPrefix(node.name.name);
```